server/src/main/java/io/crate/analyze/where/EqualityExtractor.java

```diff
@@ -122,6 +124,9 @@ private EqMatches extractMatches(Collection<ColumnIdent> columns,
         if (new ColumnsUnderNotPredicateFinder().find(query, columns)) {
             return EqMatches.NONE;
         }
+        if (new ColumnsUnderOrOperatorFinder().find(query, columns)) {
+            return EqMatches.NONE;
+        }
```

Comment:
Seems a bit weird to me that we added new post-processing logic here, when the initial pass over the query would already determine this.

Wouldn't it be possible to adjust https://github.com/crate/crate/blob/ddef9b04a188251a375aacef9f6a2411ff8a78ad/server/src/main/java/io/crate/analyze/where/EqualityExtractor.java#L426-L440 accordingly?

See also the changes in https://github.com/crate/crate/pull/10708, which initially caused the equality extractor to allow in-exact matches.